File: src/hx-admin-web/src/api-services/models/sys-job-detail.ts
import { JobCreateTypeEnum } from './job-create-type-enum';
import { BasePageInput } from './base/base-page-input';

/**
 * 
 * @export
 * @interface AddJobDetailInput
 */
 export interface AddJobDetailInput {
    
    /**
     * 组名称
     * @type {string}
     * @memberof AddJobDetailInput
     */
    groupName?: string | null;
    /**
     * 作业类型FullName
     * @type {string}
     * @memberof AddJobDetailInput
     */
    jobType?: string | null;
    /**
     * 程序集Name
     * @type {string}
     * @memberof AddJobDetailInput
     */
    assemblyName?: string | null;
    /**
     * 描述信息
     * @type {string}
     * @memberof AddJobDetailInput
     */
    description?: string | null;
    /**
     * 是否并行执行
     * @type {boolean}
     * @memberof AddJobDetailInput
     */
    concurrent?: boolean;
    /**
     * 是否扫描特性触发器
     * @type {boolean}
     * @memberof AddJobDetailInput
     */
    includeAnnotations?: boolean;
    /**
     * 额外数据
     * @type {string}
     * @memberof AddJobDetailInput
     */
    properties?: string | null;
    /**
     * 更新时间
     * @type {Date}
     * @memberof AddJobDetailInput
     */
    updatedTime?: Date | null;
    /**
     * 
     * @type {JobCreateTypeEnum}
     * @memberof AddJobDetailInput
     */
    createType?: JobCreateTypeEnum;
    /**
     * 脚本代码
     * @type {string}
     * @memberof AddJobDetailInput
     */
    scriptCode?: string | null;
    /**
     * 作业Id
     * @type {string}
     * @memberof AddJobDetailInput
     */
    jobId: string;
}


/**
 * 
 * @export
 * @interface UpdateJobDetailInput
 */
 export interface UpdateJobDetailInput extends AddJobDetailInput {
    /**
     * 雪花Id
     * @type {number}
     * @memberof UpdateJobDetailInput
     */
    id: number;
}

/**
 * 
 * @export
 * @interface DeleteJobDetailInput
 */
 export interface DeleteJobDetailInput {
    /**
     * 作业Id
     * @type {string}
     * @memberof DeleteJobDetailInput
     */
    jobId?: string | null;
}


/**
 * 
 * @export
 * @interface JobDetailInput
 */
 export interface JobDetailInput {
    /**
     * 作业Id
     * @type {string}
     * @memberof JobDetailInput
     */
    jobId?: string | null;
}

/**
 * 
 * @export
 * @interface PageJobInput
 */
 export interface PageJobInput extends BasePageInput{
    /**
     * 作业Id
     * @type {string}
     * @memberof PageJobInput
     */
    jobId?: string | null;
    /**
     * 描述信息
     * @type {string}
     * @memberof PageJobInput
     */
    description?: string | null;
}


/**
 * 系统作业信息表
 * @export
 * @interface SysJobDetail
 */
export interface SysJobDetail {
    /**
     * 雪花Id
     * @type {number}
     * @memberof SysJobDetail
     */
    id?: number;
    /**
     * 作业Id
     * @type {string}
     * @memberof SysJobDetail
     */
    jobId: string;
    /**
     * 组名称
     * @type {string}
     * @memberof SysJobDetail
     */
    groupName?: string | null;
    /**
     * 作业类型FullName
     * @type {string}
     * @memberof SysJobDetail
     */
    jobType?: string | null;
    /**
     * 程序集Name
     * @type {string}
     * @memberof SysJobDetail
     */
    assemblyName?: string | null;
    /**
     * 描述信息
     * @type {string}
     * @memberof SysJobDetail
     */
    description?: string | null;
    /**
     * 是否并行执行
     * @type {boolean}
     * @memberof SysJobDetail
     */
    concurrent?: boolean;
    /**
     * 是否扫描特性触发器
     * @type {boolean}
     * @memberof SysJobDetail
     */
    includeAnnotations?: boolean;
    /**
     * 额外数据
     * @type {string}
     * @memberof SysJobDetail
     */
    properties?: string | null;
    /**
     * 更新时间
     * @type {Date}
     * @memberof SysJobDetail
     */
    updatedTime?: Date | null;
    /**
     * 
     * @type {JobCreateTypeEnum}
     * @memberof SysJobDetail
     */
    createType?: JobCreateTypeEnum;
    /**
     * 脚本代码
     * @type {string}
     * @memberof SysJobDetail
     */
    scriptCode?: string | null;
}

File: src/starshine-admin-api/Starshine.Admin.Models/Entities/SysJobDetail.cs
﻿namespace Starshine.Admin.Models;

/// <summary>
/// 系统作业信息表
/// </summary>
[SugarTable(null, "系统作业信息表")]
public class SysJobDetail : EntityBase
{
    /// <summary>
    /// 作业Id
    /// </summary>
    [SugarColumn(ColumnDescription = "作业Id", Length = 64)]
    public string JobId { get; set; }

    /// <summary>
    /// 组名称
    /// </summary>
    [SugarColumn(ColumnDescription = "组名称",IsNullable =true, Length = 128)]
    public string? GroupName { get; set; } = "default";

    /// <summary>
    /// 作业类型FullName
    /// </summary>
    [SugarColumn(ColumnDescription = "作业类型", IsNullable = true, Length = 128)]
    public string? JobType { get; set; }

    /// <summary>
    /// 程序集Name
    /// </summary>
    [SugarColumn(ColumnDescription = "程序集", IsNullable = true, Length = 128)]
    public string? AssemblyName { get; set; }

    /// <summary>
    /// 描述信息
    /// </summary>
    [SugarColumn(ColumnDescription = "描述信息", IsNullable = true, Length = 128)]
    public string? Description { get; set; }

    /// <summary>
    /// 是否并行执行
    /// </summary>
    [SugarColumn(ColumnDescription = "是否并行执行")]
    public bool Concurrent { get; set; } = true;

    /// <summary>
    /// 是否扫描特性触发器
    /// </summary>
    [SugarColumn(ColumnDescription = "是否扫描特性触发器", ColumnName = "annotation")]
    public bool IncludeAnnotations { get; set; } = false;

    /// <summary>
    /// 额外数据
    /// </summary>
    [SugarColumn(ColumnDescription = "额外数据", IsNullable = true, ColumnDataType = StaticConfig.CodeFirst_BigString)]
    public string? Properties { get; set; } = "{}";

    /// <summary>
    /// 更新时间
    /// </summary>
    [SugarColumn(ColumnDescription = "更新时间")]
    public DateTime? UpdatedTime { get; set; }

    /// <summary>
    /// 作业创建类型
    /// </summary>
    [SugarColumn(ColumnDescription = "作业创建类型")]
    public JobCreateTypeEnum CreateType { get; set; } = JobCreateTypeEnum.BuiltIn;

    /// <summary>
    /// 脚本代码
    /// </summary>
    [SugarColumn(ColumnDescription = "脚本代码", IsNullable = true, ColumnDataType = StaticConfig.CodeFirst_BigString)]
    public string? ScriptCode { get; set; }
}